flake.nix

```diff
@@ -32,17 +32,21 @@
       };
     };
     nil-crypto3 = {
-      url = "https://github.com/NilFoundation/crypto3";
-      type = "git";
-      submodules = true;
+      type = "github";
+      owner = "NilFoundation";
+      repo = "crypto3";
+      rev = "3bd5b8df2091274abaa28fd86b9e3e89d661b95a";
```

Comment:
Why hardcode revision here. but not in the lock file?

Reply:
That's exactly the point: version which is locked in lock-file is not "pinned" and may be accidentally with `nix flake update` or even with `nix build`, if you will change something in the input specification. When you pin the version here, you can't update it without manually changing the revision, which looks like more explicit approach. The motivation here is simple: crypto3 is constantly changing and there are a lot of backward incompatible changes.